### DISCUSSION TITLE
How to find the crazyflie address?

### DISCUSSION THREAD

**Discussion by StewenceNewton** [bitcraze/discussions#1304] [Q&A]
Hi my name is newton, from IIT Hyderabad, India. I'm working on the swarm of crazyflies unfortunately i forgot the one of the drone's communication channel address. I tried bootloader mode also but still it i was not able to find the address. Can you please help me on that

**Comment by Jool99** [comment]
* Plug crazyflie into computer with USB
* Start cfclient
* Start crazyflie
* Click 'Scan'
* Select USB interface and click 'Connect'
* Select Connect>Configure 2.X in alternatives bar in top

**Reply by mattnoh** [reply]
hello I know this is a bit late but I think I am having a similar problem and when I click scan on the cfclient the usb interface does not show up. is this a known issue?

**Reply by Jool99** [reply]
Dont know. I just noticed by coincident that sometimes I have to turn on the drone after plugging it in and starting cfclient for it to show up.

**Reply by mattnoh** [reply]
hello thank you for the answer!! however even though I tried to plug it in while the drone is on or off after or before it still doesn't show up in the cfclient which is a shame

**Reply by ArisMorgens** [reply]
Hello! The issue you're experiencing seems to be caused by the lack of USB permissions. You should follow this [guide](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/installation/usb_permissions/) to be able to use the Crazyflie over USB.

**Comment by mattnoh** [comment]
![Screenshot from 2025-01-23 11-52-32](https://github.com/user-attachments/assets/db69092d-3c87-4408-b946-7b7d8747d277)

hello! Thank you for the answer! I think I have already allowed the usb permissions following the guide. here is a screenshot of my current state and I was wondering if I had missed anything as unfortunately I still cannot connect my crazyfile with an usb cable even though I can coldbootload the drone which I would think that it isnt at least bricked.

**Reply by ArisMorgens** [reply]
I suggest to put the Crazyflie into bootloader mode and then reflash the firmware in recovery mode, following these [instructions](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/). 
You should not be able to see the Crazyflie connected with USB, so for this you must use the Crazyradio.

**Reply by mattnoh** [reply]
hello thank you for the response. I have managed to put the crazyfile into the bootloader mode and went through the cold boot(recovory) several times. but with it being on the newest available firmware I still cannot get it to show up on the cfclient. I have in case tried to rollback the firmware to a past release and the same problem happens.

what is weird though is that I think I do have a brand new board in a box and when trying to connect to that one the same problem happens. I still can put it into the bootloader mode and cold boot but still cant connect it to the cfclient.   

thank you for the help

**Reply by ArisMorgens** [reply]
Hello. After the cold boot (recovery) you should be able to see the Crazyflie in the 'usb://0' channel of the cfclient. You should also be able to connect to it with the Crazyradio by using the correct address.
Could you please send the terminal output after clicking the 'scan' button? 
Maybe this way we can better understand what's going on.

**Reply by mattnoh** [reply]
hello thanks for the help. even after cold boot I could not see the crazyfile in the usb;//0 channel for a while but after using a different cable I could see the crazyfile connected to my pc as usb://0 now. and after resetting the eeprom through my pc I could now connect to the crazyfile as normal. thank you so much.

**Reply by ArisMorgens** [reply]
Really nice to hear that you managed to connect to your Crazyflie. Using a charging-only cable is a common mistake that I should have thought of. Anyway, enjoy flying your Crazyflie!
